Add tooltips to pasteable items in the bank

BlueSoapTurtle opened the pull request:
This adds a tooltip to all items which can be turned into paste, showing how much of a paste a stack gives, how much each item from the stack gives and also how much gp each past would cost by using an item.

All options are configurable in the config as well

This might cause some merge conflicts since I have another PR up for changing one of the classes but I can fix that in whichever PR is closed last

BlueSoapTurtle committed:
Add tooltips to pasteable items in the bank

This adds a tooltip to all items which can be turned into paste,
showing how much of a paste a stack gives, how much each item from the stack gives
and also how much gp each past would cost by using an item.

All options are configurable in the config as well

DapperMickie commented:
Can you add this for the inventory as well?

DapperMickie reviewed (approved):
LGTM otherwise

hex-agon commented on src/main/java/work/fking/masteringmixology/MasteringMixologyTooltips.java:
```diff
@@ -0,0 +1,139 @@
+    private final TooltipManager tooltipManager;
+    @Inject
+    private ItemManager itemManager;
+    private final StringBuilder itemStringBuilder = new StringBuilder();
```
You should not re-use `StringBuilders`, also string concatenation nowadays is the preferred approach.

BlueSoapTurtle replied:
You're right! Forgot we're not on java 8 :D

BlueSoapTurtle committed:
Fix tooltip not showing outside the bank

BlueSoapTurtle committed:
Change stringbuilder to concatenation

BlueSoapTurtle commented:
@DapperMickie Fixed the inventory tooltips!

BlueSoapTurtle commented:
Fixed the conflicts, let me know if something else needs to be done with the PR! :)

BlueSoapTurtle commented:
I should probably rebase this also..

BlueSoapTurtle commented:
Somehow messed the branch up a bit, fixed it now. PR ready for review/merge!

BlueSoapTurtle committed:
Fix code style

BlueSoapTurtle commented:
Ready for review / merge

raiyni commented:
needs rebased again

BlueSoapTurtle committed:
Merge branch 'master' into add-paste-tooltip

BlueSoapTurtle commented:
back home now so will update the PRs, just rebased it, will see if it works

BlueSoapTurtle commented:
tested and it works, was just on the layout tabs that it can't show since those give another ID, ready to merge IMO

BlueSoapTurtle committed:
Merge branch 'master' into add-paste-tooltip

BlueSoapTurtle commented:
Rebased, ready for review or merge

BlueSoapTurtle commented:
Adding some photos to show what the change does

The tooltip when hovering an item
![image](https://github.com/user-attachments/assets/3e78ef18-a856-4711-9f4c-11319a3ea73b)
Config
![image](https://github.com/user-attachments/assets/853ea717-4f92-4877-bffa-d3564a3818c3)
Enabling GE Price per paste:
![image](https://github.com/user-attachments/assets/005f5860-7e73-4192-9c71-c92515f52bb4)

BlueSoapTurtle committed:
Cleanup code

BlueSoapTurtle commented:
Pushed a change to improve the code:
- Renamed Herb to PasteIngredient
  - Store every item id in a map for quicker lookup
- Changed PotionComponent to set the formattedName and character from the enum name 
- Cleaned the MasteringMixologyTooltips
  - Merged some checks from the render method into the getItemContainer method
  - Removed the @Inject on itemManager
  - Removed the color on the tooltip as it did nothing
- Improve config fields name, description and position

Pretty happy with the code now, feel free to review or merge :)
